plugins/woocommerce-admin/client/tasks/fills/products/use-create-product-by-type.ts

```diff
@@ -38,7 +43,16 @@ export const useCreateProductByType = () => {
 			);
 
 			if ( assignment.variationName === 'treatment' ) {
-				navigateTo( { url: getNewPath( {}, '/add-product', {} ) } );
+				if ( isNewExperienceEnabled ) {
```

Comment:
Should this check be outside the treatment condition?  Meaning if the experience is already turned on, should we just navigate to the new experience instead of checking the experiment?

Reply:
Yes, that's a good idea! I changed that in the commit [0fe8dc8](https://github.com/woocommerce/woocommerce/pull/36406/commits/0fe8dc8b5deadec78409aac8565e75dc2572994b)